rp2040-hal/src/timer.rs

```diff
@@ -84,27 +94,85 @@ impl Timer {
     }
     /// Retrieve a reference to alarm 0. Will only return a value the first time this is called
     pub fn alarm_0(&mut self) -> Option<Alarm0> {
-        take_alarm(1 << 0).then_some(Alarm0(PhantomData))
+        take_alarm(1 << 0).then_some(Alarm0(*self))
     }
 
     /// Retrieve a reference to alarm 1. Will only return a value the first time this is called
     pub fn alarm_1(&mut self) -> Option<Alarm1> {
-        take_alarm(1 << 1).then_some(Alarm1(PhantomData))
+        take_alarm(1 << 1).then_some(Alarm1(*self))
     }
 
     /// Retrieve a reference to alarm 2. Will only return a value the first time this is called
     pub fn alarm_2(&mut self) -> Option<Alarm2> {
-        take_alarm(1 << 2).then_some(Alarm2(PhantomData))
+        take_alarm(1 << 2).then_some(Alarm2(*self))
     }
 
     /// Retrieve a reference to alarm 3. Will only return a value the first time this is called
     pub fn alarm_3(&mut self) -> Option<Alarm3> {
-        take_alarm(1 << 3).then_some(Alarm3(PhantomData))
+        take_alarm(1 << 3).then_some(Alarm3(*self))
+    }
+
+    /// Pauses execution for at minimum `us` microseconds.
+    fn delay_us(&self, mut us: u32) {
```

Comment:
I think this should be named `delay_us_internal` or something similar to avoid the odd reborrows `(*self).delay_us(…)`.